asset/asset.go

```diff
@@ -462,6 +464,136 @@ const (
 	ScriptV0 ScriptVersion = 0
 )
 
+// TapscriptTreeNodes represents the two supported ways to define a tapscript
+// tree to be used as a sibling for a Taproot Asset commitment, an asset group
+// key, or an asset script key. This type is used for interfacing with the DB,
+// not for supplying in a proof or key derivation. The inner fields are mutually
+// exclusive.
+type TapscriptTreeNodes struct {
+	// An ordered list of TapLeaf objects, representing a Tapscript tree.
+	leaves *TapLeafNodes
+
+	// The TapHashes of the child nodes of a TapBranch.
+	branch *TapBranchNodes
+}
+
+type TapLeafNodes struct {
+	v []txscript.TapLeaf
+}
+
+// TapBranchNodesLen is the length of a TapBranch represented as a byte arrray.
+const TapBranchNodesLen = 64
+
+type TapBranchNodes struct {
+	left  [chainhash.HashSize]byte
+	right [chainhash.HashSize]byte
+}
+
+func (t TapscriptTreeNodes) GetLeaves() fn.Option[TapLeafNodes] {
+	if t.leaves == nil {
+		return fn.None[TapLeafNodes]()
+	}
+
+	return fn.Some[TapLeafNodes](*t.leaves)
+}
+
+func (t TapscriptTreeNodes) GetBranch() fn.Option[TapBranchNodes] {
+	if t.branch == nil {
+		return fn.None[TapBranchNodes]()
+	}
+
+	return fn.Some[TapBranchNodes](*t.branch)
+}
+
+func (t TapscriptTreeNodes) FromBranch(tbn TapBranchNodes) TapscriptTreeNodes {
```

Comment:
Why is this a method on `TapscriptTreeNodes`? Same below. Those should probably just be standalone functions?

Reply:
Agree, leftover from previous approaches.